test/verify/files/mock-insights

```diff
@@ -124,6 +125,51 @@ class handler(BaseHTTPRequestHandler):
             self.wfile.write(b'{ "reports": [ "foo", "bar" ] }\n')
             return
 
+        m = self.match("/r/insights/platform/ingress/v1/upload")
+        if m:
+            # The metadata of the system is in the multipart MIME data
+            # sent by the system for the upload; to pick it and use it
+            # we need to unpack the multipart MIME data.
+
+            # First, create prologue to the data, so it can be recognized
+            # as multipart MIME.
+            multipart_data = (
+                b"""MIME-Version: 1.0
+Content-Type: """
+                + self.headers.get("content-type").encode("utf-8")
+                + b"""
+
+"""
```

Comment:
This is quite hard to read, and may be better written as `b"\n\n"`?

Reply:
hm not bad- would it be possible to fix it in a followup? this PR is a simple backport of the existing commits from sub-man-cockpit